// @faceless-ui/collapsibles/dist/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.useCollapsibleGroup = exports.useCollapsible = exports.CollapsibleToggler = exports.CollapsibleGroupContext = exports.CollapsibleGroup = exports.CollapsibleContent = exports.CollapsibleContext = exports.Collapsible = void 0;
var Provider_1 = require("./Collapsible/Provider");
Object.defineProperty(exports, "Collapsible", { enumerable: true, get: function () { return __importDefault(Provider_1).default; } });
var context_1 = require("./Collapsible/Provider/context");
Object.defineProperty(exports, "CollapsibleContext", { enumerable: true, get: function () { return __importDefault(context_1).default; } });
var Content_1 = require("./Collapsible/Content");
Object.defineProperty(exports, "CollapsibleContent", { enumerable: true, get: function () { return __importDefault(Content_1).default; } });
var Provider_2 = require("./CollapsibleGroup/Provider");
Object.defineProperty(exports, "CollapsibleGroup", { enumerable: true, get: function () { return __importDefault(Provider_2).default; } });
var context_2 = require("./CollapsibleGroup/Provider/context");
Object.defineProperty(exports, "CollapsibleGroupContext", { enumerable: true, get: function () { return __importDefault(context_2).default; } });
var Toggler_1 = require("./Collapsible/Toggler");
Object.defineProperty(exports, "CollapsibleToggler", { enumerable: true, get: function () { return __importDefault(Toggler_1).default; } });
var useCollapsible_1 = require("./Collapsible/useCollapsible");
Object.defineProperty(exports, "useCollapsible", { enumerable: true, get: function () { return __importDefault(useCollapsible_1).default; } });
var useCollapsibleGroup_1 = require("./CollapsibleGroup/useCollapsibleGroup");
Object.defineProperty(exports, "useCollapsibleGroup", { enumerable: true, get: function () { return __importDefault(useCollapsibleGroup_1).default; } });
