// @faceless-ui/slider/dist/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DotsNav = exports.DotNav = exports.withSlider = exports.useSlider = exports.SliderTrack = exports.SliderProgress = exports.SliderNav = exports.SliderButton = exports.SliderContext = exports.SliderProvider = exports.Slide = void 0;
var Slide_1 = require("./Slide");
Object.defineProperty(exports, "Slide", { enumerable: true, get: function () { return __importDefault(Slide_1).default; } });
var SliderProvider_1 = require("./SliderProvider");
Object.defineProperty(exports, "SliderProvider", { enumerable: true, get: function () { return __importDefault(SliderProvider_1).default; } });
var context_1 = require("./SliderProvider/context");
Object.defineProperty(exports, "SliderContext", { enumerable: true, get: function () { return __importDefault(context_1).default; } });
var SliderButton_1 = require("./SliderButton");
Object.defineProperty(exports, "SliderButton", { enumerable: true, get: function () { return __importDefault(SliderButton_1).default; } });
var SliderNav_1 = require("./SliderNav");
Object.defineProperty(exports, "SliderNav", { enumerable: true, get: function () { return __importDefault(SliderNav_1).default; } });
var SliderProgress_1 = require("./SliderProgress");
Object.defineProperty(exports, "SliderProgress", { enumerable: true, get: function () { return __importDefault(SliderProgress_1).default; } });
var SliderTrack_1 = require("./SliderTrack");
Object.defineProperty(exports, "SliderTrack", { enumerable: true, get: function () { return __importDefault(SliderTrack_1).default; } });
var useSlider_1 = require("./useSlider");
Object.defineProperty(exports, "useSlider", { enumerable: true, get: function () { return __importDefault(useSlider_1).default; } });
var withSlider_1 = require("./withSlider");
Object.defineProperty(exports, "withSlider", { enumerable: true, get: function () { return __importDefault(withSlider_1).default; } });
var DotNav_1 = require("./DotNav");
Object.defineProperty(exports, "DotNav", { enumerable: true, get: function () { return __importDefault(DotNav_1).default; } });
Object.defineProperty(exports, "DotsNav", { enumerable: true, get: function () { return __importDefault(DotNav_1).default; } }); // for backward compatibility
