// @faceless-ui/css-grid/dist/Grid/index.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.useGrid = void 0;
var react_1 = __importStar(require("react"));
var Cell_1 = require("../Cell");
var Settings_1 = require("../Settings");
var Context = (0, react_1.createContext)({
    cols: {
        s: 0,
        m: 0,
        l: 0,
        xl: 0,
    },
});
var useGrid = function () { return (0, react_1.useContext)(Context); };
exports.useGrid = useGrid;
var Grid = (0, react_1.forwardRef)(function (props, ref) {
    var _a, _b, _c, _d;
    var children = props.children, _e = props.htmlElement, Tag = _e === void 0 ? 'div' : _e, className = props.className, rest = __rest(props, ["children", "htmlElement", "className"]);
    var containingCell = (0, Cell_1.useCell)();
    var settings = (0, Settings_1.useSettings)();
    var classPrefix = settings.classPrefix;
    var value = {
        cols: {
            xl: ((_a = containingCell === null || containingCell === void 0 ? void 0 : containingCell.cols) === null || _a === void 0 ? void 0 : _a.xl) || settings.cols.xl,
            l: ((_b = containingCell === null || containingCell === void 0 ? void 0 : containingCell.cols) === null || _b === void 0 ? void 0 : _b.l) || settings.cols.l,
            m: ((_c = containingCell === null || containingCell === void 0 ? void 0 : containingCell.cols) === null || _c === void 0 ? void 0 : _c.m) || settings.cols.m,
            s: ((_d = containingCell === null || containingCell === void 0 ? void 0 : containingCell.cols) === null || _d === void 0 ? void 0 : _d.s) || settings.cols.s,
        },
    };
    return (react_1.default.createElement(Context.Provider, { value: value },
        react_1.default.createElement(Tag, __assign({ ref: ref, className: [
                className,
                "".concat(classPrefix, "__grid"),
                "".concat(classPrefix, "__grid--xl-cols-").concat(value.cols.xl),
                "".concat(classPrefix, "__grid--l-cols-").concat(value.cols.l),
                "".concat(classPrefix, "__grid--m-cols-").concat(value.cols.m),
                "".concat(classPrefix, "__grid--s-cols-").concat(value.cols.s),
            ].filter(Boolean).join(' ') }, rest), children)));
});
Grid.displayName = 'Grid';
exports.default = Grid;
